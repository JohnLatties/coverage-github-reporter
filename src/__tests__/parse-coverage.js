/* eslint-env jest */

const { resolve } = require('path')

const { parseFile } = require('../parse-coverage')

describe('parseFile', () => {
  it('parses', () => {
    expect(parseFile('/root/app', resolve(__dirname, 'coverage-final.json'))).toMatchSnapshot()
  })
  it('parses 2', () => {
    expect(parseFile('/root/app', resolve(__dirname, 'coverage-final2.json'))).toMatchSnapshot()
  })
})
