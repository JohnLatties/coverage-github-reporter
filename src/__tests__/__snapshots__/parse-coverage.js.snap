// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseFile parses 1`] = `
Object {
  "*": Object {
    "percent": 18.57142857142857,
  },
  "src/": Object {
    "files": Object {
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "format-coverage.js": Object {
        "percent": 36.13458528951487,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
      "parse-coverage.js": Object {
        "percent": 13.636363636363637,
      },
    },
    "percent": 18.57142857142857,
  },
}
`;

exports[`parseFile parses 2 1`] = `
Object {
  "*": Object {
    "percent": 58.705357142857146,
  },
  "src/": Object {
    "files": Object {
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "format-coverage.js": Object {
        "percent": 94.05320813771519,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
      "parse-coverage.js": Object {
        "percent": 90.58441558441558,
      },
    },
    "percent": 58.705357142857146,
  },
}
`;
